backend/average-payments.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+////    Settings
+$Database       = __DIR__.'/../data/average_payments.db';
+
+////    Parse Variables
+$Processing = false;
+if ( !empty($_GET['os']) && !empty($_GET['payment']) ) {
+    require_once __DIR__.'/authenticatron.load.php';
+    // Accept any code +/- 10 Minutes.
+    // Ignore incorrect codes.
+    if ( Authenticatron_Check($_GET['authenticatron_code'], $Secret, 20) ) {
+        $Processing = true;
+    }
+}
+if ( $Processing ) {
+    $OS = strtolower(htmlentities($_GET['os'], ENT_QUOTES, 'UTF-8'));
+    $Payment = intval($_GET['payment']);
+}
+
+////    Error Handling
+function LastError($db) {
+    $Error = 'Error Code "'.$db->lastErrorCode().'" : '.$db->lastErrorCode();
+    var_dump($Error);
+}
+
+////    Open Database
+if ( $Processing ) {
+    if ( !is_writable($Database) ) {
```

Comment:
It might be handy to also check if the folder where `$Database` is located is also writeable. If my memory serves, sqlite wants to put an lockfile in the same folder it's database file is